### ISSUE TITLE
Need to do something with own provided certificate to get management tools working in secure mode?

### ISSUE THREAD

**Issue by XsylenZ** [arrowhead-f/mgmt-tool-js#6]
Hi,

I installed the framework by compiling my own debian install files with maven. So I added the certificates into my my browser (MacOS keychain). After that I used the docker version of your management tools and at first sight it looked all good until I open one of the other functions (i.e.: Orchestrator Store, or Auth.) and after that all pages went blank.

It looks like the same issue as issue #5 the only difference between them is that I used the core system on a self installed debian machine instead of the docker one (read: own certificate).

Do I need to do something with the certificate to be able working with the management tools? Cause it looks like it can't use the ssl connection because it needs someone to approve that it  is unsafe?

**Comment by XsylenZ** [comment]
For more information, header of the main request (http://localhost:3000/registry):

![image](https://user-images.githubusercontent.com/10152270/70709525-59503400-1cdd-11ea-867e-1a498a3a1add.png)

**Comment by tsvetlin** [comment]
Hello!

In Mac OS as and Chrome, you need to to import the Certificate to your Keychain (what you did). Please make sure it is usable.

Chrome should open you a window, asking which certificate should it use.

![image](https://user-images.githubusercontent.com/15650896/70710126-d039fc80-1cde-11ea-9ab7-b059180178b4.png)

![image](https://user-images.githubusercontent.com/15650896/70710158-e47df980-1cde-11ea-88fc-94dfd0fd3ca8.png)

**Comment by XsylenZ** [comment]
Okay, this is not happening, but the certificate I added from /etc/arrowhead/system/sysop is also not green but red, as in not trusted.

![image](https://user-images.githubusercontent.com/10152270/70711208-42abdc00-1ce1-11ea-836e-005a8069cd55.png)

So probably why it is not working, is because Chrome keeps throwing me a warning screen instead.

![image](https://user-images.githubusercontent.com/10152270/70711131-0a0c0280-1ce1-11ea-8a27-42b668bef64f.png)

If the installation creates its own certificates, do I have to do something with them before I install them on my Mac?

**Comment by tsvetlin** [comment]
No, you just have to install them. But as your screenshot shows, you installed the root arrowhead certificate and not the sysop.p12.
Please install the sysop.p12 certificate.

If it is not green, please select `Always trust`
![image](https://user-images.githubusercontent.com/15650896/70711436-d7163e80-1ce1-11ea-82f8-adaffeff58bb.png)

**Comment by XsylenZ** [comment]
I did install the sysop.p12 and then it installs the root version and also the sysop one. So now I have both.

I put the root on always Trust and then it does also trust the sysop cert:

![image](https://user-images.githubusercontent.com/10152270/70711647-4ab84b80-1ce2-11ea-923e-e18bcb6ebfa3.png)

But it still throws this warning message in Chrome on Swagger and at the management tools it stays blank.

I'll try to clear the cache and let you know what happend.

**Comment by XsylenZ** [comment]
Okay I found the problem:

![image](https://user-images.githubusercontent.com/10152270/70717906-90c7dc00-1cef-11ea-93c0-c5ceb5ddf724.png)

So I use the localhost name (deb-arrowhead.local) of my debian machine, and that is not in the certificate. So the certificate only has the ip address, "localhost" and "deb-arrowhead":

![image](https://user-images.githubusercontent.com/10152270/70718386-63c7f900-1cf0-11ea-9011-19c94d98c7f1.png)

The error that Chrome gives is a bit misleading, because it's not about the common names. Chrome checks the Subject Alternative Names (SAN).

> SAN must contain proper DNS or IP entry.
>
>  - When DNS is used, it should be a resolvable FQDN name.
>  - When an IP address is used, it should be explicitely specified as such within the SAN chain.

As found in this answer: [https://serverfault.com/questions/880804/can-not-get-rid-of-neterr-cert-common-name-invalid-error-in-chrome-with-self](url)

So now I entered the ip number and Chrome stops crying about it, and also used the ip number to start the management tools from docker. Works as it should be.

Thanks for looking into this!
